frame/support/procedural/src/pallet/parse/config.rs

```diff
@@ -322,8 +342,8 @@ impl ConfigDef {
 
 			if type_attrs_const.len() == 1 {
 				match trait_item {
-					syn::TraitItem::Type(type_) => {
-						let constant = syn::parse2::<ConstMetadataDef>(type_.to_token_stream())
+					syn::TraitItem::Type(ref type_) => {
+						let constant = ConstMetadataDef::try_from(type_)
 							.map_err(|e| {
 								let error_msg = "Invalid usage of `#[pallet::constant]`, syntax \
 									must be `type $SomeIdent: Get<$SomeType>;`";
```

Comment:
I think you can remove this error map, the error message you produce is more precise.